### DISCUSSION TITLE
Issue/Clarification on "copilot env show" and "copilot job ls" commands

### DISCUSSION THREAD

**Discussion by tsingh2k15** [aws/copilot-cli#3236] [Q&A]
I am using copilot for services and jobs. I need to check if a copilot job exists in an environment but it seems like there is no support in the existing commands which would allow me to do this. I need more guidance here.

Scenario:
I am trying to deploy two copilot jobs with in same AWS account. This account will have two environments(ECS clusters). Each job will be associated to a different ECS cluster.  In my CI/CD pipeline, I need to check if a job already exists in a given environment and create it if it does not exist. 

Problem:
I could not achieve this while using `copilot env show` command as it does not list the jobs for a given environment.
I could not achieve this while using `copilot job ls` command as it does not support the environment filter.

Please share your thoughts on this. Below is the error I get when try to init the job in the same AWS account but in a different environment(ECS cluster), it says that the job already exists which is true but I intend to create the job in different ECS cluster(environment). Please let me know if this is not clear, I can add more information if needed.

`copilot job init --name some-job --job-type 'Scheduled Job' --dockerfile ./Dockerfile_some-job`

It seems like you are trying to init a job that already exists.
To recreate the job, please run:

`copilot job delete --name some-job`

To me, it appears that job commands do not support the notion of environment? Looking at the manifest, it seems to be supported there.  https://aws.github.io/copilot-cli/docs/manifest/scheduled-job/

**Comment by iamhopaul123** [comment]
Hello @tsingh2k15. That's a great finding! We do need such kind of enhancement to allow users to use `env show` to find out deployed jobs. Opened a new issue for tracking this: https://github.com/aws/copilot-cli/issues/3237

**Reply by tsingh2k15** [reply]
> 

thanks @iamhopaul123 
I look forward to using this enhancement.

**Reply by tsingh2k15** [reply]
@iamhopaul123 
On related note, is there a way to turn off the scheduled job via manifest? https://aws.github.io/copilot-cli/docs/manifest/scheduled-job/

I see an option at console UI. I know I can turn it off indirectly by setting the cron or rate to very high interval, but was wondering if I can just say something like "off" or "disable" in the manifest and then turn in on as needed.

![enable-disable](https://user-images.githubusercontent.com/25232829/152053383-15541031-9f1e-45a9-aa55-832926a743b5.png)

**Reply by iamhopaul123** [reply]
Sorry we don't have an actual way to turn it off. Our recommendation for now would be setting the past date so that it won't be triggered. See more discussion [here](https://github.com/aws/copilot-cli/issues/1823#issuecomment-759622192). Let me know if you have any suggestion!
